pkg/object/ceph.go

```diff
@@ -139,6 +139,9 @@ func (c *ceph) Put(key string, in io.Reader) error {
 		if b, ok := in.(*bytes.Reader); ok {
 			v := reflect.ValueOf(b)
 			data := v.Elem().Field(0).Bytes()
+			if len(data) == 0 {
+				return errors.New("ceph: can't put empty file")
+			}
 			return ctx.WriteFull(key, data)
 		}
 		buf := cephPool.Get().([]byte)
```

Comment:
`in` could also be empty string, we should return an error

Reply:
`ctx.WriteFull` will panic when  `len(data)=0`

Reply:
<img width="1183" alt="image" src="https://github.com/juicedata/juicefs/assets/31313340/9a6f3773-ea64-4f06-90a6-56cd61fdb44f">

Reply:
I meant we should also fix the case that it's not `*bytes.Reader`